### PR TITLE
Update asyncapi-react to version 0.3.0

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Update `asyncapi-react` to version 0.3.0
- replace yarn to npm
- removed styled-components from packages
- create playground for development process
- serialize open-api spec from yaml or string to json

**Related issue(s)**
https://github.com/kyma-project/console/issues/684
Resolve https://github.com/kyma-incubator/documentation-component/issues/38

### CODE REVIEW

File: packages/odata-react/tsconfig.json [hunk 1]
```diff
@@ -3,13 +3,8 @@
   "compilerOptions": {
     "outDir": "./lib",
     "baseUrl": "./src",
-    "allowJs": true,
-    "allowSyntheticDefaultImports": true,
-    "isolatedModules": true,
-    "noEmit": true,
-    "jsx": "preserve",
-    "declaration": false
+    "declaration": true
```
Comment (aerfio) [inline packages/odata-react/tsconfig.json]: declaration is inside tsconfig.base.json

File: packages/odata-render-engine/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "extends": "./../../tsconfig.base.json",
   "compilerOptions": {
     "outDir": "./lib",
-    "baseUrl": "./src"
+    "baseUrl": "./src",
+    "declaration": true
```
Comment (aerfio) [inline packages/odata-render-engine/tsconfig.json]: .

File: packages/open-api-render-engine/src/renderEngine.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 import { OpenApiProps } from "./types";
+
 import "swagger-ui-dist/swagger-ui.css";
 
-function createSwagger(schema: any, plugins: any) {
+function createSwagger(schema: string | any, plugins: any) {
   return import("swagger-ui-dist").then(swagger => {
     const presets = [swagger.SwaggerUIBundle.presets.apis, plugins];
 
-    // temporary any type, because some fucking idiot added not correct types to @types/swagger-ui-dist package
```
Comment (aerfio) [inline packages/open-api-render-engine/src/renderEngine.tsx]: whooops XD

File: packages/open-api-render-engine/tsconfig.json [hunk 4]
```diff
@@ -2,7 +2,8 @@
   "extends": "./../../tsconfig.base.json",
   "compilerOptions": {
     "outDir": "./lib",
-    "baseUrl": "./src"
+    "baseUrl": "./src",
+    "declaration": true
```
Comment (aerfio) [inline packages/open-api-render-engine/tsconfig.json]: .

File: packages/open-api-render-engine/src/renderEngine.tsx [hunk 5]
```diff
@@ -34,7 +34,7 @@ function prepareDataForCreate(schema: any, url: string): any {
 }
 
 export const OpenApiRenderEngine: React.FunctionComponent<
-  RenderEngineProps<OpenApiProps, any>
+  RenderEngineProps<OpenApiProps, string | any>
```
Comment (aerfio) [inline packages/open-api-render-engine/src/renderEngine.tsx]: `string | any` is `any`

File: playground/package.json [hunk 6]
```diff
@@ -11,23 +11,14 @@
...
   "scripts": {
-    "start": "SKIP_PREFLIGHT_CHECK=true react-app-rewired start",
-    "build": "react-scripts build",
-    "test": "echo \"No tests here!\"; exit 0;",
-    "eject": "react-scripts eject"
-  },
-  "eslintConfig": {
-    "extends": "react-app"
+    "start": "SKIP_PREFLIGHT_CHECK=true ../node_modules/.bin/react-app-rewired start",
+    "test": "../node_modules/.bin/react-app-rewired test"
```
Comment (aerfio) [inline playground/package.json]: why do we have test inside playgroud

Comment (aerfio) [reply playground/package.json]: and why do we have react-app-rewired installed in root, and not in playground, if we will only ever have it installed inside playground

File: playground/src/App.tsx [hunk 7]
```diff
@@ -34,7 +36,7 @@ const RENDERERS = {
 
 export const App = () => (
   <DC.Provider
-    sources={SOURCES}
+    sources={SOURCES as Sources<any>}
```
Comment (aerfio) [inline playground/src/App.tsx]: you changed playground to TS and now we use `any`?

File: playground/src/index.css [hunk 8]
```diff
@@ -1,10 +1,15 @@
...
+}
+
 body {
   margin: 0;
   font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
     "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
     sans-serif;
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
+  background: #fff !important;
```
Comment (aerfio) [inline playground/src/index.css]: why !important

File: tools/gulp/constants.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  [p: string]: string;
+} = {
+  [PACKAGES.DOCUMENTATION_COMPONENT]: "documentation-component",
+  [PACKAGES.ODATA]: "odata-react",
+  [PACKAGES.ASYNC_API_RE]: "async-api-render-engine",
+  [PACKAGES.MARKDOWN_RE]: "markdown-render-engine",
+  [PACKAGES.ODATA_RE]: "odata-render-engine",
+  [PACKAGES.OPEN_API_RE]: "open-api-render-engine",
+};
+export const packagesDirs = Object.keys(Packages).map(
```
Comment (aerfio) [inline tools/gulp/constants.ts]: use Object.values instead

File: tools/gulp/tasks/clean.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import { task, src, series } from "gulp";
+// @ts-ignore
```
Comment (aerfio) [inline tools/gulp/tasks/clean.ts]: why do you use this directive?

File: tsconfig.base.json [hunk 11]
```diff
@@ -1,22 +1,19 @@
...
   "compilerOptions": {
     "target": "es5",
     "module": "esnext",
     "lib": ["dom", "esnext"],
     "jsx": "react",
     "strict": true,
-    "alwaysStrict": true,
-    "strictFunctionTypes": true,
-    "forceConsistentCasingInFileNames": true,
+    "strictFunctionTypes": false,
```
Comment (aerfio) [inline tsconfig.base.json]: "strictFunctionTypes": true,

File: tools/gulp/gulpfile.ts [hunk 12]
```diff
@@ -0,0 +1,4 @@
+import "./tasks/clean";
```
Comment (aerfio) [inline tools/gulp/gulpfile.ts]: I don't like the fact that we are using deprecated api, and that we are using that kind of imports - this file exports all of tasks, yet we do not explicitly see anything here, it's useless

please use newer api and make those exports explicit in this file

File: packages/markdown-render-engine/package.json [hunk 13]
```diff
@@ -31,9 +31,9 @@
     "lodash.has": "^4.5.2",
     "lodash.isdate": "^4.0.1",
     "lodash.mapvalues": "^4.6.0",
-    "prism-react-renderer": "^0.1.6",
+    "prism-react-renderer": "^0.1.7",
     "react-copy-to-clipboard": "^5.0.1",
-    "react-markdown": "^4.1.0",
+    "react-markdown": "^4.2.2",
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply packages/markdown-render-engine/package.json]: Yes

File: tools/gulp/tasks/clean.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+import deleteEmpty from "delete-empty";
+// @ts-ignore
+import clean from "gulp-clean";
+
+import { PACKAGES_DIR } from "../constants";
+
+function cleanLib(done: () => void) {
```
Comment (aerfio) [inline tools/gulp/tasks/clean.ts]: extract ()=>void

File: tools/gulp/tasks/clean.ts (hunk 10 above)
Comment (aerfio) [reply tools/gulp/tasks/clean.ts]: comment that the types are old please

File: package.json [hunk 15]
```diff
@@ -1,45 +1,45 @@
...
-    "build": "npm run clean && gulp build",
-    "build:dev": "gulp build --dist node_modules/@kyma-project",
-    "prerelease": "gulp copy-misc && gulp build --dist node_modules/@kyma-project",
-    "publish:latest": "npm run prerelease && npm run build && lerna publish --dist-tag=latest -m \"Publish %s release\"",
-    "publish:rc": "npm run prerelease && npm run build && lerna publish --dist-tag=rc -m \"Publish %s release\"",
-    "publish:next": "npm run prerelease && npm run build && lerna publish --dist-tag=next --skip-git -m \"Publish %s release\"",
-    "publish:beta": "npm run prerelease && npm run build && lerna publish --dist-tag=beta -m \"Publish %s release\"",
-    "publish:test": "npm run prerelease && npm run build && lerna publish --force-publish --dist-tag=test --skip-git -m \"Publish %s release\""
+    "clean-bundles": "gulp clean-bundles",
+    "clean-package-locks": "gulp clean-package-locks",
```
Comment (aerfio) [inline package.json]: maybe it's not a good idea to have such a task made "public"

File: package.json [hunk 16]
```diff
@@ -1,45 +1,45 @@
...
-    "@babel/plugin-proposal-class-properties": "^7.5.5",
-    "@babel/plugin-proposal-object-rest-spread": "^7.5.5",
-    "@babel/plugin-transform-runtime": "^7.5.5",
-    "@babel/preset-env": "^7.5.5",
-    "@babel/preset-react": "^7.0.0",
-    "@babel/preset-typescript": "^7.3.3",
+    "@types/cli-color": "^0.3.29",
+    "@types/fancy-log": "^1.3.1",
+    "@types/gulp": "4.0.6",
+    "@types/gulp-sass": "^4.0.0",
```
Comment (aerfio) [inline package.json]: delete those types altogether if they are wrong

**Pull request merged**